tests/DrupalDoctrineCacheTest.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+namespace Capgemini\CacheTest;
+
+define('CACHE_PERMANENT', 0);
+
+/**
+ * @class DrupalDoctrineCacheTest
+ * @package Capgemini\CacheTest
+ */
+class DrupalDoctrineCacheTest extends \PHPUnit_Framework_TestCase {
+
+  /**
+   * @covers Capgemini\Cache\DrupalDoctrineCache::setCacheTable
+   * @covers Capgemini\Cache\DrupalDoctrineCache::getCacheTable
+   */
+  public function testSetCacheTable() {
+    $cache = new DrupalDoctrineCacheStub();
+    $cache->setCacheTable('cache_test');
+    $this->assertEquals('cache_test', $cache->getCacheTable());
+  }
+
+  /**
+   * @covers Capgemini\Cache\DrupalDoctrineCache::doSave
+   * @covers Capgemini\Cache\DrupalDoctrineCache::getCacheId
+   */
+  public function testCacheSet() {
+    $cache = new DrupalDoctrineCacheStub();
+    $cache->save('key', 'value');
+    $this->assertEquals(1, $cache->getCallCount('cache_set:doctrine:[key][1],value,cache,*', TRUE));
+    $cache->save('key', 'value', NULL);
+    $this->assertEquals(2, $cache->getCallCount('cache_set:doctrine:[key][1],value,cache,0'));
+    $cache->save('key', 'value', 60);
+    $this->assertEquals(1, $cache->getCallCount('cache_set:doctrine:[key][1],value,cache,' . substr(time(), 0, 1) . '*', TRUE));
```

Comment:
I looked into this, but I'm not sure it makes the test any simpler. We have different save() parameters and different assertions in each case.

```diff
diff --git a/tests/DrupalDoctrineCacheTest.php b/tests/DrupalDoctrineCacheTest.php
index 7ca9dbb..cee800f 100644
--- a/tests/DrupalDoctrineCacheTest.php
+++ b/tests/DrupalDoctrineCacheTest.php
@@ -20,18 +20,35 @@ class DrupalDoctrineCacheTest extends \PHPUnit_Framework_TestCase {
     $this->assertEquals('cache_test', $cache->getCacheTable());
   }
 
+  public function cacheSetProvider() {
+    return array(
+      array('key', 'value'),
+      array('key', 'value', NULL),
+      array('key', 'value', 60),
+    );
+  }
+
   /**
+   * @dataProvider cacheSetProvider
    * @covers Capgemini\Cache\DrupalDoctrineCache::doSave
    * @covers Capgemini\Cache\DrupalDoctrineCache::getCacheId
    */
-  public function testCacheSet() {
+  public function testCacheSet($key, $value, $expiry = 'unset') {
     $cache = new DrupalDoctrineCacheStub();
-    $cache->save('key', 'value');
-    $this->assertEquals(1, $cache->getCallCount('cache_set:doctrine:[key][1],value,cache,*', TRUE));
-    $cache->save('key', 'value', NULL);
-    $this->assertEquals(2, $cache->getCallCount('cache_set:doctrine:[key][1],value,cache,0'));
-    $cache->save('key', 'value', 60);
-    $this->assertEquals(1, $cache->getCallCount('cache_set:doctrine:[key][1],value,cache,' . substr(time(), 0, 1) . '*', TRUE));
+    if ($expiry == 'unset') {
+      $cache->save($key, $value);
+      $expiry = NULL;
+    }
+    else if ($expiry || $expiry == NULL) {
+      $cache->save($key, $value, $expiry);
+    }
+
+    if ($expiry) {
+      $this->assertEquals(1, $cache->getCallCount('cache_set:doctrine:[key][1],value,cache,' . substr(time(), 0, 1) . '*', TRUE));
+    }
+    else {
+      $this->assertEquals(1, $cache->getCallCount('cache_set:doctrine:[key][1],value,cache,0'));
+    }
   }
 
   /**
```